src/ReturnTypes/ModelExtension.php

```diff
@@ -109,6 +110,10 @@ public function getTypeFromStaticMethodCall(
             $returnType = new GenericObjectType(EloquentBuilder::class, [new ObjectType($scope->resolveName($methodCall->class))]);
         }
 
+        if (in_array(Collection::class, $returnType->getReferencedClasses()) && $methodCall->name->name === 'all') {
+            $returnType = new IntersectionType([new ObjectType(\Illuminate\Support\Collection::class), $returnType->getTypes()[1]]);
```

Comment:
It should be the generic type here. `new GenericType(Collection::class, [new ObjectType()])`